packaging/os/xbps.py

```diff
@@ -0,0 +1,301 @@
+            else:
+                module.fail_json(msg="Could not upgrade")
+    else:
+        module.fail_json(msg="Could not upgrade")
+
+
+def remove_packages(module, xbps_path, packages):
+    """Returns true if package removal succeeds"""
+    remove_c = 0
+    # Using a for loop incase of error, we can report the package that failed
+    for package in packages:
+        # Query the package first, to see if we even need to remove
+        installed, updated = query_package(module, xbps_path, package)
+        if not installed:
+            continue
+
+        cmd = "%s -y %s" % (xbps_path['remove'], package)
+        rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+
+        if rc != 0:
+            module.fail_json(msg="failed to remove %s" % (package))
+
+        remove_c += 1
+
+    if remove_c > 0:
+
+        module.exit_json(changed=True, msg="removed %s package(s)" % remove_c)
+
+    module.exit_json(changed=False, msg="package(s) already absent")
+
+
+def install_packages(module, xbps_path, state, packages):
+    """Returns true if package install succeeds."""
+    toInstall = []
+    for i, package in enumerate(packages):
+        """If the package is installed and state == present or state == latest
+        and is up-to-date then skip"""
+        installed, updated = query_package(module, xbps_path, package)
+        if installed and (state == 'present' or
+                          (state == 'latest' and updated)):
+            continue
+
+        toInstall.append(package)
+
+    if len(toInstall) == 0:
+        module.exit_json(changed=False, msg="Nothing to Install")
+
+    cmd = "%s -y %s" % (xbps_path['install'], " ".join(toInstall))
+    rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+
+    if rc != 0 and not (state == 'latest' and rc == 17):
+        module.fail_json(msg="failed to install %s" % (package))
+
+    module.exit_json(changed=True, msg="installed %s package(s)"
+                     % (len(toInstall)))
+
+    module.exit_json(changed=False, msg="package(s) already installed")
+
+
+def check_packages(module, xbps_path, packages, state):
+    """Returns change status of command"""
+    would_be_changed = []
+    for package in packages:
+        installed, updated = query_package(module, xbps_path, package)
+        if ((state in ["present", "latest"] and not installed) or
+                (state == "absent" and installed) or
+                (state == "latest" and not updated)):
+            would_be_changed.append(package)
+    if would_be_changed:
+        if state == "absent":
+            state = "removed"
+        module.exit_json(changed=True, msg="%s package(s) would be %s" % (
+            len(would_be_changed), state))
+    else:
+        module.exit_json(changed=False, msg="package(s) already %s" % state)
+
+
+def main():
+    """Returns, calling appropriate command"""
+
+    mask = os.umask(0)  # fix the umask
```

Comment:
A problem was observed with an early version of this module wherein the package database would be altered by a command run by the module and would no longer by accessible by normal users.  Setting the umask to 0 allows the code from libxbps to handle the permissions and enforce them as necessary.